recipes/default.rb

```diff
@@ -19,58 +19,11 @@
 # limitations under the License.
 #
 
-include_recipe 'sysctl::service'
+::Chef::Recipe.send(:include, SysctlCookbook::SysctlHelpers::Param)
 
-directory 'Sysctl config directory' do
-  path node['sysctl']['conf_dir']
-  owner 'root'
-  group 'root'
-  mode 0o755
-  action :create
-  only_if { node['sysctl']['conf_dir'] }
-end
-
-sysctl_config_file = Sysctl.config_file(node)
-
-if sysctl_config_file
-  # If default sysctl.params attributes are not set, set them at recipe compile time
-  # to the values output by the last run. This allows the LWRPs to act idempotently
-  if File.exist?(sysctl_config_file)
-    File.read(sysctl_config_file).lines.each do |l|
-      next unless l =~ /^[\w\.]+?=/
-      key, val = l.chomp.split('=')
-      key_path = key.split('.')
-      location = key_path.slice(0, key_path.size - 1).reduce(node.default['sysctl']['params']) do |m, o|
-        m[o] ||= {}
-        m[o]
-      end
-      location[key_path.last] ||= val.to_s
-    end
-  end
-
-  # this is called by the sysctl_param lwrp to trigger template creation
-  ruby_block 'save-sysctl-params' do
-    action :nothing
-    block do
-    end
-    notifies :create, "template[#{sysctl_config_file}]", :delayed
-  end
-
-  # this is called by the sysctl::apply recipe to trigger template creation
-  ruby_block 'apply-sysctl-params' do
-    action :nothing
-    block do
-    end
-    notifies :create, "template[#{sysctl_config_file}]", :immediately
-  end
-
-  # this needs to have an action in case node.sysctl.params has changed
-  # and also needs to be called for persistence on lwrp changes via the
-  # ruby_block
-  template sysctl_config_file do
-    action :nothing
-    source 'sysctl.conf.erb'
-    mode '0644'
-    notifies :restart, 'service[procps]', :immediately
+coerce_attributes(node['sysctl']['params']).each do |x|
+  k, v = x.split('=')
+  sysctl_param k do
+    value v
   end
-end
+end if node['sysctl']['params']
```

Comment:
On a bare node, with the recipe in the `run_list` but without any sysctl params, this fail with NoMethodError `undefined method '[]' for nil:NilClass`

We can change this condition to something like that :)

```ruby
end if node['sysctl']
```